pkg/config/config.go

```diff
@@ -568,6 +568,23 @@ type ImageConfig struct {
 	// containing credentials necessary for pulling images from secure
 	// registries.
 	GlobalAuthFile string `toml:"global_auth_file"`
+	// NamespacedAuthDir is the root path for pod namespace-separated
+	// auth files, which is intended to be used together with CRI-O's credential provider:
+	// https://github.com/cri-o/credential-provider
+	// The final auth file will be <NAMESPACED_AUTH_DIR>/<NAMESPACE>-<IMAGE_NAME_SHA256>.json.
+	// This temporary auth file will be used instead of any configured
+	// GlobalAuthFile. The file will be automatically removed by CRI-O
+	// after its usage on image pull. This means that there are theoretically
+	// race conditions possible within a single namespace, if the same image
+	// get's pulled at the same time. The kubelet will avoid those races
+	// by managing which images should be pulled in parallel.
```

Comment:
@QiWang19 @bitoku I did a bit more research and saw that we have an outstanding issue when the kubelet parallel image pull feature is being used: https://github.com/kubernetes/kubernetes/issues/125164